Frameworks/Foundation/NSOperation.mm

```diff
@@ -84,11 +84,32 @@ - (void)_checkReady {
     }
 }
 
+- (void)_completionBlockThread {
+    decltype(_completionBlock) localCompletion;
+
+    { // _completionBlockLock scope
+        std::lock_guard<std::recursive_mutex> lock(_completionBlockLock);
+        localCompletion = std::move(_completionBlock);
```

Comment:
`std::move` is well-defined for `AutoId`: the source is nil'd.

That said: a moved-from object is said to be in an undefined state. If we give it a defined state, we are attempting to trump the standard.

Thoughts?